src/test/scala/mesosphere/marathon/api/MarathonExceptionMapperTest.scala

```diff
@@ -18,12 +22,36 @@ class MarathonExceptionMapperTest extends MarathonSpec with GivenWhenThen with M
 
     Then("The correct response is created")
     response.getStatus should be(400)
-    val entity = response.getEntity.asInstanceOf[Map[String, AnyRef]]
-    entity("message") should be("Invalid JSON")
-    val details = entity("details").asInstanceOf[List[Map[String, Seq[String]]]]
+    val entityString = response.getEntity.asInstanceOf[String]
+    val entity = Json.parse(entityString)
+    (entity \ "message").as[String] should be("Invalid JSON")
+    val details = (entity \ "details").as[Seq[JsObject]]
     details should have size 1
-    details.head("path") should be("/id")
-    details.head("errors") should have size 1
-    details.head("errors").head should be("error.expected.jsstring")
+    val firstDetail = details.head
+    (firstDetail \ "path").as[String] should be("/id")
+    val errors = (firstDetail \ "errors").as[Seq[String]]
+    errors should have size 1
+    errors.head should be("error.expected.jsstring")
+  }
+
+  test("Render ConstraintValidationException correctly") {
+    Given("A ConstraintValidationException from an invalid app")
+    val violations = BeanValidation.validate(V2AppDefinition(id = PathId("/test")))
+    val ex = intercept[ConstraintViolationException] { BeanValidation.requireValid(violations) }
+    val mapper = new MarathonExceptionMapper()
+
+    When("The mapper creates a response from this exception")
+    val response = mapper.toResponse(ex)
+
+    Then("The correct response is created")
+    response.getStatus should be(422)
+    val entityString = response.getEntity.asInstanceOf[String]
+    val entity = Json.parse(entityString)
+    (entity \ "message").as[String] should be("Bean is not valid")
```

Comment:
I know that this is the current error as well, but "bean" is a java concept that we shouldn't leak to the users.